nrfu_tests/test_base_services_ntp_association.py

```diff
@@ -53,49 +66,56 @@ def test_ntp_clocks(self, dut, tests_definitions):
+                tests_tools.post_process_skip(tops, self.test_ntp_clocks, self.output)
+                pytest.skip(tops.output_msg)
 
             ntp_association_cmd = "show ntp associations"
 
             """
             TS: Running `show ntp associations` on device and verifying that the
-            primary and secondary NTP association is correct on the host.
+            expected NTP association is correct on the host.
             """
             output = tops.run_show_cmds([ntp_association_cmd])
             logging.info(
                 f"On device {tops.dut_name} output of {ntp_association_cmd} command is:\n{output}\n"
             )
             self.output += f"Output of {ntp_association_cmd} command is: \n{output}"
             ntp_associations = output[0]["result"].get("peers")
-            assert ntp_associations, "NTP association details are not collected."
+            assert (
+                ntp_associations
+            ), "NTP association details are not collected in the command output."
 
-            for peer in ntp_associations:
-                if ntp_associations[peer].get("condition") == "sys.peer":
+            for _, peer_details in ntp_associations.items():
+                if peer_details.get("condition") == "sys.peer":
                     tops.actual_output["primary_ntp_association"] = "sys.peer"
-                elif ntp_associations[peer].get("condition") == "candidate":
-                    tops.actual_output["secondary_ntp_association"] = "candidate"
+                if not single_ntp_check:
+                    if peer_details.get("condition") == "candidate":
+                        tops.actual_output["secondary_ntp_association"] = "candidate"
 
             # Forming an output message if a test result is fail
             if tops.actual_output != tops.expected_output:
                 tops.output_msg = ""
-                not_primary_ntp_association = (
-                    tops.actual_output["primary_ntp_association"]
-                    != tops.expected_output["primary_ntp_association"]
-                )
-                not_secondary_ntp_association = (
-                    tops.actual_output["secondary_ntp_association"]
-                    != tops.expected_output["secondary_ntp_association"]
-                )
-                if not_primary_ntp_association and not_secondary_ntp_association:
-                    tops.output_msg += (
-                        "Primary and Secondary NTP associations are not found on the device.\n"
-                    )
-                elif not_primary_ntp_association:
+                not_primary_ntp_association = tops.actual_output.get(
+                    "primary_ntp_association"
+                ) != tops.expected_output.get("primary_ntp_association")
+                not_secondary_ntp_association = tops.actual_output.get(
+                    "secondary_ntp_association"
+                ) != tops.expected_output.get("secondary_ntp_association")
```

Comment:
Can we write it as below to make it more readable.
```suggestion
                actutal_primary_ntp_status = tops.actual_output.get("primary_ntp_association")
                expected_primary_ntp_status = tops.expected_output.get("primary_ntp_association")
                actual_secondary_ntp_status =  tops.actual_output.get("secondary_ntp_association")
                expected_secondary_ntp_status =  tops.expected_output.get("secondary_ntp_association")
                
                no_primary_ntp = actutal_primary_ntp_status != expected_primary_ntp_status
                no_secondary_ntp = actual_secondary_ntp_status  != expected_secondary_ntp_status 
```